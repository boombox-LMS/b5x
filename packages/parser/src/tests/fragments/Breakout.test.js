const {
  exampleBreakoutMarkupOne,
  exampleBreakoutMarkupTwo,
  exampleBreakoutMarkupThree,
} = require("../../../dist/componentBuilders/fragments/Breakout");
const { RawFragmentSchema } = require("@b5x/types");

describe("The simple breakout fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleBreakoutMarkupOne
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed simple breakout fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed simple breakout fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});

describe("The breakout fragment builds as expected when it includes an icon", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleBreakoutMarkupTwo
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed breakout fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed breakout fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});

describe("The breakout fragment builds as expected when it includes all attributes", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleBreakoutMarkupThree
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed breakout fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed breakout fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
