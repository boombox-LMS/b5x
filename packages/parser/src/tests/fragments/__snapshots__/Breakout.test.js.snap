// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The breakout fragment builds as expected when it includes all attributes The parsed breakout fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Breakout",
  "contents": "",
  "data": Object {
    "color": "#7694e4",
    "icon": "http://localhost:8080/topic-assets/mock-topic-uri/images/cat.svg",
    "iconSize": "large",
    "title": "Advice from a cat",
  },
  "dependencyUris": Array [],
  "digest": "4f9cced00fc5f6cbf94e3bf1eaab3280",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;

exports[`The breakout fragment builds as expected when it includes an icon The parsed breakout fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Breakout",
  "contents": "",
  "data": Object {
    "color": undefined,
    "icon": "https://api.iconify.design/mdi/lightbulb-on-10.svg",
    "iconSize": "small",
    "title": "Tip",
  },
  "dependencyUris": Array [],
  "digest": "20f22bbd088045d7076380b375a1f359",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;

exports[`The simple breakout fragment builds as expected The parsed simple breakout fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Breakout",
  "contents": "",
  "data": Object {
    "color": undefined,
    "iconSize": "small",
    "title": "Tip",
  },
  "dependencyUris": Array [],
  "digest": "ef716ba07b3d50254a48a500079c08ec",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
